catalog/core/catalog-core-commands/src/main/java/org/codice/ddf/commands/catalog/DuplicateCommands.java

```diff
@@ -155,23 +182,39 @@ protected void duplicateInBatches(
               0L,
               TimeUnit.MILLISECONDS,
               blockingQueue,
-              StandardThreadFactoryBuilder.newThreadFactory("duplicateCommandsThread"),
               rejectedExecutionHandler);
       console.printf("Running a maximum of %d threads during replication.%n", multithreaded);
 
-      printProgressAndFlush(start, totalWanted, ingestedCount.get());
+      printProgressAndFlush(
+          start, Math.max(totalWanted, initialMetacards.size()), ingestedCount.get());
       int index;
-      while ((index = queryIndex.addAndGet(batchSize)) <= totalWanted) {
-        final int i = index;
+      while (!done.get()) {
+        index = queryIndex.addAndGet(batchSize);
+        final int taskIndex = index;
 
         executorService.submit(
             () -> {
-              final SourceResponse response =
-                  query(queryFacade, filter, i, getQuerySizeFromIndex(totalWanted, i));
-              if (response != null) {
-                ingestMetacards(ingestFacade, getMetacardsFromSourceResponse(response));
+              int querySize = (int) getQuerySizeFromIndex(totalWanted, taskIndex);
+              if (querySize < 1) {
+                // If we don't need any more metacards, we're finished
+                done.set(true);
+                return;
```

Comment:
❓Why is this a `return` here instead of `break`? Might the `executorService.shutdown();` be skipped?